File: contacthub-android/app/src/main/java/id/my/hubkontak/utils/Telegram.java
package id.my.hubkontak.utils;

import android.content.Context;
import android.net.Uri;
import android.util.Log;
import android.widget.Toast;

import com.android.volley.DefaultRetryPolicy;
import com.android.volley.Request;
import com.android.volley.RequestQueue;
import com.android.volley.Response;
import com.android.volley.RetryPolicy;
import com.android.volley.VolleyError;
import com.android.volley.toolbox.JsonObjectRequest;
import com.android.volley.toolbox.Volley;

import org.json.JSONObject;

import static id.my.hubkontak.utils.DBContract.SOCKET_TIMEOUT;
import static id.my.hubkontak.utils.DBContract.TELEGRAM_BOT_API_KEY;
import static id.my.hubkontak.utils.DBContract.TELEGRAM_CHAT_ID;

public class Telegram {

    private static final String TAG = "Telegram";
    private final Context context;

    public Telegram(Context context) {
        this.context = context;
    }
    public void send_chat_id(String chatId,String message){
        final RequestQueue requestQueue = Volley.newRequestQueue(this.context);
        final String url = Uri.parse("https://api.telegram.org/bot" + TELEGRAM_BOT_API_KEY + "/sendmessage")
                .buildUpon()
                .appendQueryParameter("chat_id",chatId)
                .appendQueryParameter("text",message)
                .toString();

        final JsonObjectRequest jsonObjectRequest = new JsonObjectRequest(Request.Method.POST, url, null, new Response.Listener<JSONObject>() {
            @Override
            public void onResponse(JSONObject response) {
            }
        }, new Response.ErrorListener() {
            @Override
            public void onErrorResponse(VolleyError error) {
                Toast.makeText(context, error.getMessage(), Toast.LENGTH_SHORT).show();
                Log.e(TAG,error.getMessage());
            }
        });
        RetryPolicy retryPolicy = new DefaultRetryPolicy(SOCKET_TIMEOUT, DefaultRetryPolicy.DEFAULT_MAX_RETRIES, DefaultRetryPolicy.DEFAULT_BACKOFF_MULT);
        jsonObjectRequest.setRetryPolicy(retryPolicy);
        requestQueue.add(jsonObjectRequest);
    }
    public void send(String message){
        send_chat_id(TELEGRAM_CHAT_ID,message);
    }
}

File: contacthub-api/libraries/telegram.py
from flask import current_app
import requests
from datetime import datetime

class Telegram():
    def __init__(self):
        self.config = current_app.config['TELEGRAM']
        
    def send(self,text:str):
        url = 'https://api.telegram.org/bot' + self.config['TOKEN'] +  '/sendmessage?'
        for chat_id in self.config['CHAT_ID']:
            params = {
                'chat_id' : chat_id,
                'text' : text,
            }
            result = requests.post(url,params=params)
        return result
    
    def send_log(self,message:str,request):
        try:
            template = request.method + ' ' + request.url + '\n\n'
            template = template + 'datetime : \n' +datetime.now().strftime("%Y-%m-%d %H:%M:%S") +  '\n\n'
            template = template + 'message : \n' + message + '\n\n'
            template = template + 'headers : \n' + str(request.headers) + '\n\n'
            template = template + 'data : \n' + str(request.data) + '\n\n'
            template = template + 'body : \n' + str(request.get_json()) + '\n\n'
            template = template + 'params : \n' + str(request.args)
            return self.send(template)
        except Exception as e:
            print(str(e))